components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java

```diff
@@ -473,31 +595,179 @@ public void testSetCookieNonExistCookieConfig() {
+    }
+
+    @Test
+    public void testGetUserNameProvisioningUIUrlNonEmptyURL() {
+
+        String dummyURL = "Dummy_Provisioning_URL";
+        IdentityConfigParser.getInstance().getConfiguration()
+                .put("JITProvisioning.UserNameProvisioningUI", dummyURL);
+
+        String userNameProvisioningUIUrl = FrameworkUtils.getUserNameProvisioningUIUrl();
+        assertEquals(userNameProvisioningUIUrl, "/accountrecoveryendpoint/register.do");
+    }
+
+    private void setMockedSessionContextCache() {
+
+        mockStatic(SessionContextCache.class);
+        when(SessionContextCache.getInstance()).thenReturn(mockedSessionContextCache);
+    }
+
+    @Test
+    public void testGetSessionContextFromCacheNullCacheEntry() {
+
+        setMockedSessionContextCache();
+        SessionContext sessionContext = FrameworkUtils.getSessionContextFromCache(DUMMY_CACHE_KEY);
+        assertNull(sessionContext);
+    }
+
+    @Test
+    public void testGetSessionContextFromCacheValidCacheEntry() {
+
+        cacheEntry.setContext(context);
+        setMockedSessionContextCache();
+        when(mockedSessionContextCache.getValueFromCache(cacheKey)).thenReturn(cacheEntry);
+
+        SessionContext sessionContext = FrameworkUtils.getSessionContextFromCache(DUMMY_CACHE_KEY);
+        assertEquals(sessionContext, context);
+    }
+
+    @Test
+    public void testGetSessionContextFromCacheExpiredSession() throws FrameworkException {
+
+        cacheEntry.setContext(context);
+        setMockedSessionContextCache();
+        when(mockedSessionContextCache.getValueFromCache(cacheKey)).thenReturn(cacheEntry);
+        when(mockedSessionContextCache.isSessionExpired(any(SessionContextCacheKey.class),
+                any(SessionContextCacheEntry.class))).thenReturn(true);
+        IdentityEventService identityEventService = new IdentityEventServiceImpl(Collections.EMPTY_LIST, 1);
+
+        FrameworkServiceDataHolder.getInstance().setIdentityEventService(identityEventService);
+        AuthenticationContext authenticationContext = new AuthenticationContext();
+        SessionContext sessionContext = FrameworkUtils.getSessionContextFromCache(request,
+                authenticationContext, DUMMY_CACHE_KEY);
+        assertNull(sessionContext);
+    }
+
+    @Test
+    public void testGetSessionContextFromCacheNotExpiredSession() throws FrameworkException {
+
+        cacheEntry.setContext(context);
+        setMockedSessionContextCache();
+        when(mockedSessionContextCache.getSessionContextCacheEntry(cacheKey)).thenReturn(cacheEntry);
+        when(mockedSessionContextCache.isSessionExpired(any(SessionContextCacheKey.class),
+                any(SessionContextCacheEntry.class))).thenReturn(false);
+        SessionContext sessionContext = FrameworkUtils.getSessionContextFromCache(request,
+                authenticationContext, DUMMY_CACHE_KEY);
+        assertEquals(sessionContext, context);
+
```

Comment:
Unnecessary new line

Reply:
Addressed with commit c4e56d45a2072535df672c98b16637ada8cad476 in https://github.com/wso2/carbon-identity-framework/pull/3774 PR